File: snippets/swift/BreezSDKExamples/Sources/ListPayments.swift
import BreezSDKLiquid
import Foundation

func getPayment(sdk: BindingLiquidSdk) -> Payment? {
    // ANCHOR: get-payment
    let paymentHash = "<payment hash>"
    let paymentByHash = try? sdk.getPayment(
        req: GetPaymentRequest.paymentHash(paymentHash: paymentHash)
    )

    let swapId = "<swap id>"
    let paymentBySwapId = try? sdk.getPayment(
        req: GetPaymentRequest.swapId(swapId: swapId)
    )
    // ANCHOR_END: get-payment
    return paymentBySwapId
}

func listPayments(sdk: BindingLiquidSdk) -> [Payment]? {
    // ANCHOR: list-payments
    let payments = try? sdk.listPayments(req: ListPaymentsRequest())
    // ANCHOR_END: list-payments
    return payments
}

func ListPaymentsFiltered(sdk: BindingLiquidSdk) -> [Payment]? {
    // ANCHOR: list-payments-filtered
    let payments = try? sdk.listPayments(
        req: ListPaymentsRequest(
            filters: [.send],
            fromTimestamp: 1696880000,
            toTimestamp: 1696959200,
            offset: 0,
            limit: 50
        ))
    // ANCHOR_END: list-payments-filtered
    return payments
}

func ListPaymentsDetailsAddress(sdk: BindingLiquidSdk) -> [Payment]? {
    // ANCHOR: list-payments-details-address
    let address = "<Bitcoin address>"
    let payments = try? sdk.listPayments(
        req: ListPaymentsRequest(
            details: ListPaymentDetails.bitcoin(address: address)
        ))
    // ANCHOR_END: list-payments-details-address
    return payments
}

func ListPaymentsDetailsDestination(sdk: BindingLiquidSdk) -> [Payment]? {
    // ANCHOR: list-payments-details-destination
    let destination = "<Liquid BIP21 or address>"
    let payments = try? sdk.listPayments(
        req: ListPaymentsRequest(
            details: ListPaymentDetails.liquid(assetId: nil, destination: destination)
        ))
    // ANCHOR_END: list-payments-details-destination
    return payments
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ListPayments.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*
class ListPayments {
    fun getPayment(sdk: BindingLiquidSdk) {
        // ANCHOR: get-payment
        try {
            val paymentHash = "<payment hash>";
            val paymentByHash = sdk.getPayment(GetPaymentRequest.PaymentHash(paymentHash))

            val swapId = "<swap id>";
            val paymentBySwapId = sdk.getPayment(GetPaymentRequest.SwapId(swapId))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: get-payment
    }

    fun listPayments(sdk: BindingLiquidSdk) {
        // ANCHOR: list-payments
        try {
            val payments = sdk.listPayments(ListPaymentsRequest())
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments
    }

    fun listPaymentsFiltered(sdk: BindingLiquidSdk) {
        // ANCHOR: list-payments-filtered
        try {
            val payments = sdk.listPayments(
                ListPaymentsRequest(
                    listOf(PaymentType.SEND),
                    fromTimestamp = 1696880000,
                    toTimestamp = 1696959200,
                    offset = 0u,
                    limit = 50u
                ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments-filtered
    }

    fun listPaymentsDetailsAddress(sdk: BindingLiquidSdk) {
        // ANCHOR: list-payments-details-address
        try {
            val address = "<Bitcoin address>"
            val payments = sdk.listPayments(
                ListPaymentsRequest(
                    details = ListPaymentDetails.Bitcoin(address)
                ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments-details-address
    }

    fun listPaymentsDetailsDestination(sdk: BindingLiquidSdk) {
        // ANCHOR: list-payments-details-destination
        try {
            val destination = "<Liquid BIP21 or address>"
            val payments = sdk.listPayments(
                ListPaymentsRequest(
                    details = ListPaymentDetails.Liquid(assetId = null, destination = destination)
                ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments-details-destination
    }
}
